modules/collection/programs/hyprpanel.nix

```
@@ -11,44 +19,40 @@ in {
 
     package = mkPackageOption pkgs "hyprpanel" {};
 
+    hyprland.enable = mkEnableOption "Enable Hyprland integration";
+
     settings = mkOption {
-      type = attrs;
+      type = jsonType;
       default = {};
       example = {
-        bar = {
-          autoHide = "never";
-          location = "top";
-        };
-        theme = {
-          name = "catppuccin-mocha";
-          font = "JetBrains Mono Nerd Font";
+        "bar.layouts" = {
+          "0" = {
+            left = [ "dashboard" "workspaces" "windowtitle" ];
+            middle = [ "media" ];
+            right = [ "volume" "network" "bluetooth" "battery" "systray" "clock" "notifications" ];
+          };
         };
       };
       description = ''
-        Configuration written to {file}`$HOME/.config/hyprpanel/config.json`.
-        Refer to [HyprPanel documentation](https://hyprpanel.com/configuration).
+        JSON-style configuration for HyprPanel, written to
+        {file}`$HOME/.config/hyprpanel/hyprpanel.conf`.
+         Refer to [HyprPanel documentation](https://hyprpanel.com/getting_started/installation.html).
       '';
     };
-
-    config = mkOption {
-      type = nullOr (either path (either str attrs));
-      default = null;
-      description = "Alternative config input (overrides settings if set)";
-    };
   };
 
   config = mkIf cfg.enable {
-    packages = [ cfg.package ];
-
-    files.".config/hyprpanel/config.json".text = mkIf (cfg.settings != {} || cfg.config != null) (
-      let
-        configContent =
-          if cfg.config != null then
-            if lib.isString cfg.config then cfg.config
-            else if lib.isPath cfg.config then builtins.readFile cfg.config
-            else builtins.toJSON cfg.config
-          else toJSON cfg.settings;
-      in configContent
-    );
+    packages = [cfg.package];
+
+    # Integration with Hyprland exec-once
+    hj.rum.programs.hyprland.settings.exec-once = mkIf (cfg.hyprland.enable && config.hj.rum.programs.hyprland.enable) [
```

```suggestion
rum.programs.hyprland.settings.exec-once = mkIf (cfg.hyprland.enable && config.rum.programs.hyprland.enable) [
```

This could have been deduced from the way the configuration for this module is taken here:
https://github.com/snugnug/hjem-rum/blob/94ef55d05fe9b214df2d33b3a625d28657a55f02/modules/collection/programs/hyprpanel.nix#L15

Also, as mentioned previously, the second check for the hyprland module is unnecessary, leaving us with the following:

```suggestion
rum.programs.hyprland.settings.exec-once = mkIf cfg.hyprland.enable [
```